toolchain/driver/BUILD

```diff
@@ -13,10 +15,66 @@ filegroup(
     data = glob(["testdata/**/*.carbon"]),
 )
 
+# Build rules to construct the install data tree for the Carbon toolchain.
```

Comment:
This seems to be a free-standing comment, tied to a couple rules in this file -- but it's not clear where those rules end, and driver-related rules start. Putting it at the top of the file like this implies it applies to _all_ rules below, particularly given the length. Should this be in its own BUILD file so that it's clearer what's for directly structure?

Reply:
I see you move this into a separate directly in #3993. How much extra cost would it be to do that in this PR?